ml4c3/hyperoptimizers.py

```diff
@@ -1,577 +1,113 @@
-        param_lists,
-        aucs,
-    )
-    for col, dtype in trial_metrics_and_params_df.dtypes.items():
-        if dtype == float:
-            trial_metrics_and_params_df[col] = trial_metrics_and_params_df[col].apply(
-                lambda x: f"{x:.3}",
-            )
-    metric_and_param_path = os.path.join(
-        figure_path,
-        "metrics_and_hyperparameters.csv",
-    )
-    trial_metrics_and_params_df.to_csv(metric_and_param_path)
-    logging.info(f"Saved metric and hyperparameter table to {metric_and_param_path}")
-    labels = [
-        _trial_metric_and_param_label(
-            i,
-            all_losses,
-            histories,
-            trials,
-            param_lists,
-            aucs,
-        )
-        for i in range(len(trials.trials))
-    ]
-    for i, label in enumerate(labels):
-        plt.text(i, lplot[i], label, color=colors[i])
-    plt.xlabel("Iterations")
-    plt.ylabel("Losses")
-    plt.ylim(min(lplot) * 0.95, max(lplot) * 1.05)
-    plt.title("Hyperparameter Optimization\n")
-    plt.axhline(
-        cutoff,
-        label=f"Loss display cutoff at {cutoff:.3f}",
-        color="r",
-        linestyle="--",
-    )
-    loss_path = os.path.join(figure_path, "loss_per_trial" + image_ext)
-    plt.legend()
-    plt.savefig(loss_path)
-    plt.close()
-    logging.info(f"Saved loss plot to {loss_path}")
-
-    _, [ax1, ax3, ax2] = plt.subplots(
-        nrows=1,
-        ncols=3,
-        figsize=(60, 20),
-        sharey="all",
-        gridspec_kw={"width_ratios": [2, 1, 2]},
-    )
-    cm = plt.get_cmap("gist_rainbow")
-    ax1.set_xlabel("Epoch")
-    ax1.set_ylabel("Training Loss")
-    ax2.set_xlabel("Epoch")
-    ax2.set_ylabel("Validation Loss")
-    linestyles = "solid", "dotted", "dashed", "dashdot"
-    for i, history in enumerate(histories):
-        color = cm(i / len(histories))
-        training_loss = np.clip(history["loss"], a_min=-np.inf, a_max=cutoff)
-        val_loss = np.clip(history["val_loss"], a_min=-np.inf, a_max=cutoff)
-        label = labels[i]
-        ax1.plot(training_loss, label=label, linestyle=linestyles[i % 4], color=color)
-        ax1.text(len(training_loss) - 1, training_loss[-1], str(i))
-        ax2.plot(val_loss, label=label, linestyle=linestyles[i % 4], color=color)
-        ax2.text(len(val_loss) - 1, val_loss[-1], str(i))
-    ax1.axhline(
-        cutoff,
-        label=f"Loss display cutoff at {cutoff:.3f}",
-        color="k",
-        linestyle="--",
-    )
-    ax1.set_title("Training Loss")
-    ax2.axhline(
-        cutoff,
-        label=f"Loss display cutoff at {cutoff:.3f}",
-        color="k",
-        linestyle="--",
-    )
-    ax2.set_title("Validation Loss")
-    ax3.legend(
-        *ax2.get_legend_handles_labels(),
-        loc="upper center",
-        fontsize="x-small",
-        mode="expand",
-        ncol=5,
+    df = df.reset_index()
+    df.to_csv(
+        os.path.join(base_output_folder, "metrics_and_hyperparameters.csv"),
```

Comment:
Minor request, style preference:
```suggestion
        os.path.join(base_output_folder, "metrics-and-hyperparameters.csv"),
```